apps/src/templates/lessonOverview/StyledCodeBlock.jsx

```diff
@@ -12,23 +14,48 @@ export const buildProgrammingExpressionMarkdown = function(
   return `[${block}](${programmingExpression.link})`;
 };
 
-export default class StyledCodeBlock extends Component {
-  static propTypes = {
-    programmingExpression: PropTypes.shape({
-      color: PropTypes.string,
-      syntax: PropTypes.string.isRequired,
-      link: PropTypes.string,
-      parameters: PropTypes.array
-    }).isRequired
-  };
+export default function StyledCodeBlock({programmingExpression}) {
+  const blockRef = createRef();
 
-  render() {
-    const {programmingExpression} = this.props;
+  useEffect(() => {
+    if (programmingExpression.blockName && blockRef.current) {
+      const blocksDom = parseElement(
+        `<block type='${programmingExpression.blockName}' />`
+      );
+      const blockSpace = Blockly.BlockSpace.createReadOnlyBlockSpace(
+        blockRef.current,
+        blocksDom,
+        {
+          noScrolling: true,
+          inline: true
+        }
+      );
+      shrinkBlockSpaceContainer(blockSpace, true);
```

Comment:
does this only shrink, or also expand? could be worth testing with a large block

Reply:
`shrinkBlockSpaceContainer` seems to be misnamed and really does a resize ([code](https://github.com/code-dot-org/code-dot-org/blob/de8c6b100a486062056d645d41124e6209c6afb5/apps/src/templates/instructions/utils.js#L63))